implementations/rust/ockam/ockam_api/src/authenticator/direct/direct_authenticator.rs

```diff
@@ -23,11 +24,48 @@ pub type DirectAuthenticatorResult<T> = Either<T, DirectAuthenticatorError>;
 
 pub struct DirectAuthenticator {
     members: Arc<dyn AuthorityMembersRepository>,
+    account_authority: Option<AccountAuthorityInfo>,
+}
+#[derive(Clone)]
+pub struct AccountAuthorityInfo {
+    identities_attributes: Arc<IdentitiesAttributes>,
```

Comment:
I think that shouldn't be a dependency here. I' would like to refactor it. WDYT?

Reply:
sure.   I needed some way to access to the identity attributes (attested by the account credential).   But agree that it's subtle.. if one isn't careful can end up being confused between this and the member table